src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallAppsTask.groovy

```diff
@@ -56,13 +73,37 @@ class InstallAppsTask extends AbstractTask {
             throw new GradleException("The project archive was not found and cannot be installed.")
         }
         Files.copy(archive.toPath(), new File(getServerDir(project), "/" + project.liberty.installapps.appsDirectory + "/" + getArchiveName(archive.getName())).toPath(), StandardCopyOption.REPLACE_EXISTING)
+    
+        validateAppConfig(getArchiveName(archive.getName()), getBaseName(archive))
     }
     
-    private String getInstallAppPackages() {
-        if (project.plugins.hasPlugin("ear")) {
-            project.liberty.installapps.installAppPackages = "project"
+    private void validateAppConfig(String fileName, String artifactId) throws Exception {
+        String appsDir = project.liberty.installapps.appsDirectory
+        
+        if(appsDir.equalsIgnoreCase('apps') && !isAppConfiguredInSourceServerXml(fileName)){
+            applicationXml.createApplicationElement(fileName, artifactId)
+        }
+        else if(appsDir.equalsIgnoreCase('dropins') && isAppConfiguredInSourceServerXml(fileName)){
+            throw new GradleException("The application is configured in the server.xml and the plug-in is configured to install the application in the dropins folder. A configured application must be installed to the apps folder.")
         }
-        return project.liberty.installapps.installAppPackages
+    }
+    
+    protected boolean isAppConfiguredInSourceServerXml(String fileName) {
+        boolean configured = false; 
+        if (project.liberty.configFile != null && project.liberty.configFile.exists()) {
```

Comment:
So this depends on whether or not a user has set the `configFile` value along with the `configDirectory`. If it's not set then it will look for the file in the default location. Otherwise, it will get set to the user's value but either way the value won't be used unless the `configDirectory` doesn't have a server.xml file. I think I need to add something for the case where `configFile` isn't set regardless of whether or not the `configDirectory` is being used. It might be a good idea to just get the file from the `serverDir` rather than a different directory because that is the file that will actually be used and we won't need to worry about the values of `configFile` or `configDirectory`. This would have to be called after we copy the files though.